### ISSUE TITLE
BUG: small logic error in the os.walk, edits set mid iteration

### ISSUE THREAD

**Issue by Anthony-Mckale** [feeltheajf/trufflehog3#34]
When two exclude matches are next to each other, 2nd one can be missed, due to editting the target of the "in" mid flight

small fragment that proves the issue: trufflehog3\trufflehog3\source.py (refactored from diriter)

```
def dirfilter(rel: Path, dirnames: Iterable[str], exclude_set: set):
    """filter dirnames for exclude set

    Examples
    --------
    Basic usage examples

    >>> dirfilter(Path("."), ["node_modules"], ["node_modules"])
    []
    >>> dirfilter(Path("."), ["first_match", "second_match"], ["first_match", "second_match"])
    [] // FAILS as second_match is missed due the remove
    """
    for directory in dirnames:
        dirname = rel / directory
        pattern = _match(dirname, exclude_set)
        if pattern:
            log.debug(f"skipping directory '{dirname}': '{pattern}'")
            dirnames.remove(directory)
    return dirnames
```

**Comment by Anthony-Mckale** [comment]
pushing pull request now

for me, the issue meant my big node_modules was being scanned for a project, which cost about a extra 100 seconds

dead easy fix, just iterated over a clone

```for directory in dirnames:```

became

```for directory in dirnames[:]:```

pre bug fix

```
Riversafe@DESKTOP-GPVF0RR MINGW64 /c/dev/repos/eze-management-console (develop)
$ eze tools run trufflehog
trufflehog scan took a long time (155.47s), you can often speed up trufflehog significantly by excluding dependency folders like node_modules
Eze report results:

TOOL REPORT: trufflehog (scan duration: 155.5 seconds)
    total: 0
    ignored: 0


Report 'console' took 0.0 seconds
(eze-core)
```

post bug fix
```
Riversafe@DESKTOP-GPVF0RR MINGW64 /c/dev/repos/eze-management-console (develop)
$ eze tools run trufflehog
Eze report results:

TOOL REPORT: trufflehog (scan duration: 2.7 seconds)
    total: 0
    ignored: 0


Report 'console' took 0.0 seconds
(eze-core)
```

**Comment by feeltheajf** [comment]
Hey @Anthony-Mckale 

Thanks for the heads up, this should be fixed in upcoming 3.0.3
